### PR TITLE
miracode: init at 1.0

### PR DESCRIPTION
**Pull request by Coca162**
## Description of changes

[Miracode](https://github.com/IdreesInc/Miracode) is a vectorized version of the [Monocraft](https://github.com/IdreesInc/Monocraft) font. 

Since this is my first contribution I am also adding myself to maintainer list!

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/mi/miracode/package.nix [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  inherit version;
+
+  src = fetchurl {
+    url = "https://github.com/IdreesInc/Miracode/releases/download/v${version}/Miracode.ttf";
+    hash = "sha256-Q+/D/TPlqOt779qYS/dF7ahEd3Mm4a4G+wdHB+Gutmo=";
+  };
+
+  sourceRoot = ".";
+  unpackCmd = ''cp "$curSrc" $(basename $curSrc)'';
+
```
Comment (misuzu) [inline pkgs/by-name/mi/miracode/package.nix:17]:
```suggestion
  dontUnpack = true;
```

File: pkgs/by-name/mi/miracode/package.nix [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  dontConfigure = true;
+  dontBuild = true;
+
+  installPhase = ''
+    runHook preInstall
+    install -Dm644 -t $out/share/fonts/truetype *.ttf
```
Comment (misuzu) [inline pkgs/by-name/mi/miracode/package.nix:23]:
```suggestion
    mkdir -p $out/share/fonts/truetype
    ln -s $src $out/share/fonts/truetype/Miracode.ttf
```

**Review by misuzu — CHANGES_REQUESTED** [review]
Welcome! I've added some suggestions, but LGTM otherwise.

**Review by misuzu — APPROVED** [review]

File: pkgs/by-name/mi/miracode/package.nix [hunk 3]
```diff
@@ -0,0 +1,33 @@
+{ stdenv, lib, fetchurl }:
+
+let
+  version = "1.0";
+in
+stdenv.mkDerivation {
```
Comment (Aleksanaa) [inline pkgs/by-name/mi/miracode/package.nix:6]:
```suggestion
stdenvNoCC.mkDerivation {
```

File: pkgs/by-name/mi/miracode/package.nix [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+  };
+
+  dontUnpack = true;
+  dontConfigure = true;
+  dontBuild = true;
+
+  installPhase = ''
+    runHook preInstall
+    mkdir -p $out/share/fonts/truetype
+    ln -s $src $out/share/fonts/truetype/Miracode.ttf
```
Comment (Aleksanaa) [inline pkgs/by-name/mi/miracode/package.nix:22]:
```suggestion
    install -Dm644 $src $out/share/fonts/truetype/Miracode.ttf
```

(No need to mkdir)

Don't use `ln`, as we don't want the final package to depend on `src`

**Review by RaghavSood — APPROVED** [review]

**Comment by Hellodan1** [comment]
excellent work coca162

**Pull request merged**